ui-framework-documentation.md

```diff
@@ -76,34 +89,85 @@ sub provide {
 }
 ```
 
-main.pm then calls a scheduled [Test Module](#test-module) that is using
-the Distribution to access its components through Controller layer.
+main.pm calls a scheduled [Test Module](#test-module) using [declarative scheduling](https://github.com/os-autoinst/os-autoinst-distri-opensuse/blob/master/declarative-schedule-doc.md)
+(please don't schedule modules in main*.pm directly !) and uses the [Distribution](https://github.com/os-autoinst/os-autoinst-distri-opensuse/tree/master/lib/Distribution)
```

Comment:
s/and uses the/and the test module uses the specific Distribution assigned by the factory pattern to call the corresponding controller for that distribution and utilize its business logic methods/

Reply:
I re-phrased it